physicsnemo/utils/sdf.py

```diff
@@ -112,41 +135,32 @@ def signed_distance_field(
     >>> signed_distance_field(mesh_vertices, mesh_indices, input_points)
     array([0.5], dtype=float32)
     """
-    wp.init()
-
-    # Get the current warp device string (e.g., "cuda:2")
-    try:
-        device = str(wp.get_device())
-    except Exception:
-        device = "cuda"
+    if return_cupy is None:
+        return_cupy = any(
+            isinstance(arr, cp.ndarray)
+            for arr in (mesh_vertices, mesh_indices, input_points)
+        )
 
-    # If cupy arrays come in, we have to convert them to numpy arrays:
-    return_cupy = False
-    if isinstance(mesh_vertices, cp.ndarray):
-        return_cupy = True
-    if isinstance(mesh_indices, cp.ndarray):
-        return_cupy = True
-    if isinstance(input_points, cp.ndarray):
-        return_cupy = True
-
-    # Convert numpy to warp arrays:
-    mesh_vertices = wp.array(mesh_vertices, dtype=wp.vec3, device=device)
-    mesh_indices = wp.array(mesh_indices, dtype=wp.int32, device=device)
+    wp.init()
+    device = wp.get_device()
 
-    sdf_points = wp.array(input_points, dtype=wp.vec3, device=device)
+    mesh = wp.Mesh(
+        points=wp.array(mesh_vertices, dtype=wp.vec3f, device=device),
+        indices=wp.array(mesh_indices, dtype=wp.int32, device=device),
+    )
 
-    mesh = wp.Mesh(mesh_vertices, mesh_indices)
+    N = len(input_points)
 
-    sdf = wp.zeros(shape=sdf_points.shape, dtype=wp.float32, device=device)
-    sdf_hit_point = wp.zeros(shape=sdf_points.shape, dtype=wp.vec3f, device=device)
-    sdf_hit_point_id = wp.zeros(shape=sdf_points.shape, dtype=wp.int32, device=device)
+    sdf = wp.empty(shape=(N,), dtype=wp.float32, device=device)
+    sdf_hit_point = wp.empty(shape=(N,), dtype=wp.vec3f, device=device)
+    sdf_hit_point_id = wp.empty(shape=(N,), dtype=wp.int32, device=device)
```

Comment:
Another comment on these tensors too.  They are the output data, which has to be converted from warp.  On the CPU, I don't care, but on the GPU it makes more sense to initialize them in the host framework.  For example:
- cupy inputs -> cupy outputs, init these in cupy and cast to warp via dlpack memory sharing.
- torch inputs -> torch outputs (yeah, we should support this too probably), init in torch and transfer to wp with memory sharing again.

It's not totally an idle point.  Warp and Cupy and Torch all use different memory management and none of them are cudaMalloc directly.  They're all caching memory managers.  So using the "native" memory management makes more sense.

On cpu it doesn't matter.